moja-kuhinjica-frontend/src/components/mobileMenu/index.tsx

```diff
@@ -8,34 +8,75 @@ import reservationIcon from '../../../public/static/assets/images/reservationIco
 import aboutUsIcon from '../../../public/static/assets/images/aboutUsIcon.svg'
 import myReservations from '../../../public/static/assets/images/myReservations.svg'
 import editProfile from '../../../public/static/assets/images/editProfile.svg'
-import logout from '../../../public/static/assets/images/logout.svg'
+import logoutIcon from '../../../public/static/assets/images/logout.svg'
 import profile from '../../../public/static/assets/images/profileHeader.svg'
 
 import { useRouter } from 'next/router'
+import UserService from '@/service/User.service'
 interface IMenuProps {
     closeMenu: () => void
+    loggedIn?: boolean
+    setLoggedIn?: (param: boolean) => void
 }
-const Menu = ({ closeMenu }: IMenuProps): JSX.Element => {
+interface LoggedInUser {
+    id: number
+    name: string
+    surname: string
+    phoneNumber: string
+    role: string
+}
+
+const Menu = ({
+    closeMenu,
+    loggedIn,
+    setLoggedIn,
+}: IMenuProps): JSX.Element => {
+    const [user, setUser] = useState<LoggedInUser>()
     const router = useRouter()
 
+    useEffect(() => {
+        if (loggedIn) fetchLoggedInUser()
+    }, [loggedIn])
+
+    const fetchLoggedInUser = (): void => {
+        const res = UserService.getLoggedInUser()
+            .then((res) => {
+                setUser(res.data)
+            })
+            .catch((err) => {
+                console.log(err)
+            })
+    }
+
     const navigate = (url: string): void => {
         closeMenu()
         router.push(url)
     }
+
+    const logout = (): void => {
+        localStorage.removeItem('token')
+        setLoggedIn?.(false)
```

Comment:
Why do we have ? in this line? Also why is function maybe undefined?

Reply:
Function is passed as optional prop. This menu component is used both for users that are logged in and for those who are not.